server/pulp/server/async/scheduler.py

```diff
@@ -317,15 +320,58 @@ def tick(self):
         :return:    number of seconds before the next tick should run
         :rtype:     float
         """
-        ret = super(Scheduler, self).tick()
         self._failure_watcher.trim()
 
+        # Setting the celerybeat name
+        celerybeat_name = SCHEDULER_WORKER_NAME + "@" + str(celery_name) + "@" + platform.node()
+
         # this is not an event that gets sent anywhere. We process it
         # immediately.
         scheduler_event = {'timestamp': time.time(),
                            'type': 'scheduler-event',
-                           'hostname': ("%s@%s" % (SCHEDULER_WORKER_NAME, platform.node()))}
+                           'hostname': celerybeat_name}
         worker_watcher.handle_worker_heartbeat(scheduler_event)
+
+        _logger.debug(_('Lock updated by celerybeat_name %(celerybeat_name)s')
+                      % {'celerybeat_name': celerybeat_name})
+
+        old_timestamp = datetime.utcnow() - timedelta(seconds=CELERYBEAT_OLD_TIMESTAMP)
+
+        # Updating the current lock if lock is on this instance of celerybeat
+        result = CeleryBeatLock.objects(celerybeat_name=celerybeat_name).\
+            update(set__lock_timestamp=datetime.utcnow())
+
+        # If current instance has lock and updated lock_timestamp, call super
+        if result == 1:
+            ret = super(Scheduler, self).tick()
```

Comment:
L#436 - L#349 are the same as L#364 - L#367. Make this a `staticmethod` and having the same code called from two places.